moveit_core/kinematic_constraints/src/utils.cpp

```diff
@@ -273,258 +276,258 @@ moveit_msgs::msg::Constraints constructGoalConstraints(const std::string& link_n
-{
-  for (XmlRpc::XmlRpcValue::iterator it = params.begin(); it != params.end(); ++it)
-  {
-    if (it->first == "type")
-      continue;
-    else if (it->first == "weight")
-      constraint.weight = parseDouble(it->second);
-    else if (it->first == "target_radius")
-      constraint.target_radius = parseDouble(it->second);
-    else if (it->first == "target_pose")
-    {
-      if (!constructPoseStamped(it, constraint.target_pose))
-        return false;
-    }
-    else if (it->first == "cone_sides")
-      constraint.cone_sides = static_cast<int>(it->second);
-    else if (it->first == "sensor_pose")
-    {
-      if (!constructPoseStamped(it, constraint.sensor_pose))
-        return false;
-    }
-    else if (it->first == "max_view_angle")
-      constraint.max_view_angle = parseDouble(it->second);
-    else if (it->first == "max_range_angle")
-      constraint.max_range_angle = parseDouble(it->second);
-    else
-    {
-      ROS_WARN_STREAM_NAMED(LOGNAME, "orientation constraint contains unknown entity '" << it->first << "'");
-    }
-  }
-
-  constraint.sensor_view_direction = moveit_msgs::msg::VisibilityConstraint::SENSOR_X;
-
-  return true;
-}
-
-static bool collectConstraints(XmlRpc::XmlRpcValue& params, moveit_msgs::msg::Constraints& constraints)
-{
-  if (params.getType() != XmlRpc::XmlRpcValue::TypeArray)
-  {
-    ROS_ERROR_NAMED(LOGNAME, "expected constraints as array");
-    return false;
-  }
-
-  for (int i = 0; i < params.size(); ++i)
-  {
-    if (!params[i].hasMember("type"))
-    {
-      ROS_ERROR_NAMED(LOGNAME, "constraint parameter does not specify its type");
-    }
-    else if (params[i]["type"] == "joint")
-    {
-      constraints.joint_constraints.emplace_back();
-      if (!constructConstraint(params[i], constraints.joint_constraints.back()))
-        return false;
-    }
-    else if (params[i]["type"] == "position")
-    {
-      constraints.position_constraints.emplace_back();
-      if (!constructConstraint(params[i], constraints.position_constraints.back()))
-        return false;
-    }
-    else if (params[i]["type"] == "orientation")
-    {
-      constraints.orientation_constraints.emplace_back();
-      if (!constructConstraint(params[i], constraints.orientation_constraints.back()))
-        return false;
-    }
-    else if (params[i]["type"] == "visibility")
-    {
-      constraints.visibility_constraints.emplace_back();
-      if (!constructConstraint(params[i], constraints.visibility_constraints.back()))
-        return false;
-    }
-  }
-
-  return true;
-}
-
-bool constructConstraints(XmlRpc::XmlRpcValue& params, moveit_msgs::msg::Constraints& constraints)
-{
-  if (!isStruct(params, { "name", "constraints" }, "Parameter"))
-    return false;
-
-  constraints.name = static_cast<std::string>(params["name"]);
-  return collectConstraints(params["constraints"], constraints);
-}
+// TODO: Is not getting called from anywhere, check if is necessary
```

Comment:
This is intended to be used by application code. It enables the user to define a set of constraints in a config file and use it for both the generation of an approximated constraint manifold and the sampling of that manifold. Without this code those parameters have to be defined multiple times and may get out of sync. The tutorial on approximated constraint manifolds has some more information: https://ros-planning.github.io/moveit_tutorials/doc/planning_with_approximated_constraint_manifolds/planning_with_approximated_constraint_manifolds_tutorial.html

I believe @v4hn is working on documenting this feature better